### ISSUE TITLE
How did you train net.cluster_projector() ?

### ISSUE THREAD

**Issue by ArctonLin** [Armanfard-Lab/C3#1]
I have a question after watch the code.
 
The gradient of loss is only work on net.instance_projector() and net.resnet() branch.
 
It seems train net.cluster_projector() didn't update while training.
 
 
The evaulation accuracy on CIFAR-10 is about 77% (your pre-train) and after 20 epochs training it decrease to 71%.
 
So I think it is cause by the cluster-level network didn't update??
 
Am I wrong?

**Comment by Armanfard-Lab** [comment]
Hi. We will get back to you soon. 

 

From: CardLin ***@***.***> 
Sent: Thursday, February 16, 2023 1:01 AM
To: Armanfard-Lab/C3 ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [Armanfard-Lab/C3] How did you train net.cluster_projector() ? (Issue #1)

 

I have a question after watch the code.

The gradient of loss is only work on net.instance_projector() and net.resnet() branch.

It seems train net.cluster_projector() didn't update while training.

The evaulation accuracy on CIFAR-10 is about 77% (your pre-train) and after 20 epochs training it decrease to 71%.

So I think it is cause by the cluster-level network didn't update??

Am I wrong?
